### PR TITLE
Changes to address issue #93 on OAuth support

### PR DESCRIPTION
**Pull request by SensibleWood**
Summary of changes:

* Pulled in OAuth objects and samples from OpenAPI specification
* Also pulled in OpenID Connect discovery (as this is very useful, although there's some questions on scope that need to be address IMO - for another time...)
* Updated JSON schema with new objects
* Added relevant objects to test doc
* Blew out the list of supported security standards into a bullet point list - to aid readability
* Added package-lock.json

### CODE REVIEW

File: schema/asyncapi.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
-  "title": "AsyncAPI 1.2.0 schema.",
-  "id": "http://asyncapi.hitchhq.com/v1/schema.json#",
+  "title": "AsyncAPI 2.0.0 schema.",
+  "id": "http://asyncapi.hitchhq.com/v2/schema.json#",
```
Comment (fmvilas) [inline schema/asyncapi.json]: Let's not raise the version here. I'll do it when releasing a new version. BTW, is there any breaking change in this PR?

Comment (SensibleWood) [reply schema/asyncapi.json]: This should be non-breaking as I've only added new objects Actually thought of another useful change to the schema on the OAuth flow objects so I've bring that in as well when I correct the version number

Comment (SensibleWood) [reply schema/asyncapi.json]: Changes added as request - please see updates in pull request

File: README.md [hunk 2]
```diff
@@ -1922,10 +1932,101 @@ scheme: bearer
...
+```
 
+#### <a name="oauthFlowsObject"></a>OAuth Flows Object
 
+Allows configuration of the supported OAuth Flows.
+
+##### Fixed Fields
+Field Name | Type | Description
+---|:---:|---
+<a name="oauthFlowsImplicit"></a>implicit| [OAuth Flow Object](#oauthFlowObject) | Configuration for the OAuth Implicit flow
```
Comment (mattheworiordan) [inline README.md]: A meaningless comment I know, but should you not be consistent with spacing i.e. 
```
<a name="oauthFlowsImplicit"></a>implicit | [OAuth Flow Object](#oauthFlowObject) | Configuration for the OAuth Implicit flow 
```
if you want to be consistent with the table heading columns and most other columns (apart from `---|:---:|---`)

Comment (SensibleWood) [reply README.md]: 👍 actually a cut-and-paste from the OpenAPI spec - but I'll check for it in future

**Pull request merged**